ivm/src/interact.rs

```diff
@@ -17,13 +17,24 @@ impl<'ivm, 'ext> IVM<'ivm, 'ext> {
   /// Link two ports.
   pub fn link(&mut self, a: Port<'ivm>, b: Port<'ivm>) {
     use Tag::*;
-    match (a.tag(), b.tag()) {
-      (Wire, _) => self.link_wire(unsafe { a.as_wire() }, b),
-      (_, Wire) => self.link_wire(unsafe { b.as_wire() }, a),
-      sym!(Global | Erase) | sym!(ExtVal | Erase) => self.stats.erase += 1,
-      sym!(Comb) | sym!(ExtFn) if a.label() == b.label() => self.active_fast.push((a, b)),
-      sym!(Global, _) | sym!(Comb | ExtFn | Branch) => self.active_slow.push((a, b)),
-      sym!(Erase, _) | sym!(ExtVal, _) => self.active_fast.push((a, b)),
+    match ((a.tag(), a), (b.tag(), b)) {
+      ((Wire, a), (_, b)) => self.link_wire(unsafe { a.as_wire() }, b),
+      ((_, a), (Wire, b)) => self.link_wire(unsafe { b.as_wire() }, a),
+      sym!((Global | Erase, _)) => self.stats.erase += 1,
+      sym!((ExtVal, n), (Erase, _)) => {
```

Comment:
Instead, keep the `match` as it was `(a.tag(), b.tag())` and have an arm `sym!(ExtVal | Erase)` and check `if a.tag() == ExtVal && !a.is_copy || b.tag() == ExtVal && !b.is_copy() { .. }` and do the flag-setting there.